otherlibs/ocamlc_loc/src/lexer.mll

```diff
@@ -0,0 +1,127 @@
+{
+  (* raise when the format is unrecognized *)
+  exception Unknown_format
+
+  type lines =
+    | Single of int
+    | Range of int * int
+
+  type source =
+    | Code of { code : int ; name : string }
+    | Alert of string
+
+  type severity =
+    | Error of source option
+    | Warning of source
+
+  type loc =
+    { chars : (int * int) option
+    ; lines : lines
+    ; path : string
+    }
+
+  type line = { indent : int ; contents : string }
+
+  type token =
+    | Toplevel of
+      { indent : int
+      ; loc : loc
+      ; severity : severity
+      ; message : string
+      }
+    | Related of { indent : int ; loc : loc ; message: string }
+    | Line of line
+    | Eof
+
+  let parse_range s =
+    match String.split_on_char '-' s with
+    | [x; y] -> (int_of_string x, int_of_string y)
+    | _ -> assert false
+}
+
+let blank = [' ' '\t' '\r']*
+
+let newline = '\n'
+
+let quoted_string = '"' [^ '"']+ '"'
```

Comment:
This negated pattern might end up swallowing new lines, if you care about that sort of thing.

Reply:
I suppose it's alright. The bigger problem is that we don't support quotes as part of the filename. But that's fine with me :)